### PR TITLE
Heltec ESP32 Series boards update

### PR DESCRIPTION
**Pull request by Heltec-Aaron-Lee**
## Description of Change

Fixes:

- Fixes some basic information about previous development boards;
- Fixes the option for LoRaWAN parameter in the Tools menu;
- Remove useless parameter options;
- Correcting typing issues.

Add:
- Add Heltec new boards to the development framework;
- Add external 32K Crystal oscillator option for LoRawWAN.

### CODE REVIEW

File: variants/heltec_capsule_sensor_v3/pins_arduino.h [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+#include "soc/soc_caps.h"
+
+#define HELTEC_CAPSULE_SENSOR_V3 true
+
+#define USB_VID 0x303a
+#define USB_PID 0x1001
+
+#define EXTERNAL_NUM_INTERRUPTS 46
+#define NUM_DIGITAL_PINS        48
+#define NUM_ANALOG_INPUTS       20
```
Comment (P-R-O-C-H-Y) [inline variants/heltec_capsule_sensor_v3/pins_arduino.h:14]: This is no longer required, as it's defined globally per SoC. 
```suggestion
```

File: variants/heltec_capsule_sensor_v3/pins_arduino.h [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+// and change this setup for the chosen pin (for example 38)
+static const uint8_t LED_BUILTIN = SOC_GPIO_PIN_COUNT+48;
+#define BUILTIN_LED  LED_BUILTIN // backward compatibility
+#define LED_BUILTIN LED_BUILTIN
+#define RGB_BUILTIN LED_BUILTIN
+#define RGB_BRIGHTNESS 64
+
+#define analogInputToDigitalPin(p)  (((p)<20)?(analogChannelToDigitalPin(p)):-1)
+#define digitalPinToInterrupt(p)    (((p)<48)?(p):-1)
+#define digitalPinHasPWM(p)         (p < 46)
```
Comment (P-R-O-C-H-Y) [inline variants/heltec_capsule_sensor_v3/pins_arduino.h:27]: This is no longer required, as its defined globally per SoC.
```suggestion
```

File: variants/heltec_ht_de01/pins_arduino.h [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+#define HT_DE01	true
+
+#define EXTERNAL_NUM_INTERRUPTS 16
+#define NUM_DIGITAL_PINS        40
+#define NUM_ANALOG_INPUTS       16
+
+#define analogInputToDigitalPin(p)  (((p)<20)?(analogChannelToDigitalPin(p)):-1)
+#define digitalPinToInterrupt(p)    (((p)<40)?(p):-1)
+#define digitalPinHasPWM(p)         (p < 34)
```
Comment (P-R-O-C-H-Y) [inline variants/heltec_ht_de01/pins_arduino.h:14]: Same here, this is no longer required, as its defined globally per SoC.
```suggestion
```

File: variants/heltec_wireless_mini_shell/pins_arduino.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
+#ifndef Pins_Arduino_h
+#define Pins_Arduino_h
+
+#include <stdint.h>
+#include "soc/soc_caps.h"
+
+#define WIRELESS_MINI_SHELL true
+#define EXTERNAL_NUM_INTERRUPTS 22
+#define NUM_DIGITAL_PINS        22
+#define NUM_ANALOG_INPUTS       6
```
Comment (P-R-O-C-H-Y) [inline variants/heltec_wireless_mini_shell/pins_arduino.h:10]: Same here, this is no longer required, as its defined globally per SoC.
```suggestion
```

File: variants/heltec_wireless_mini_shell/pins_arduino.h [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+static const uint8_t LED_BUILTIN = SOC_GPIO_PIN_COUNT+8;
+#define BUILTIN_LED  LED_BUILTIN // backward compatibility
+#define LED_BUILTIN LED_BUILTIN
+#define RGB_BUILTIN LED_BUILTIN
+#define RGB_BRIGHTNESS 64
+
+#define analogInputToDigitalPin(p)  (((p)<NUM_ANALOG_INPUTS)?(analogChannelToDigitalPin(p)):-1)
+#define digitalPinToInterrupt(p)    (((p)<NUM_DIGITAL_PINS)?(p):-1)
+#define digitalPinHasPWM(p)         (p < EXTERNAL_NUM_INTERRUPTS)
```
Comment (P-R-O-C-H-Y) [inline variants/heltec_wireless_mini_shell/pins_arduino.h:20]: Same here, this is no longer required, as its defined globally per SoC.
```suggestion
```

File: boards.txt [hunk 6]
```diff
@@ -21227,7 +21228,7 @@ heltec_wifi_kit_32.build.target=esp32
 heltec_wifi_kit_32.build.mcu=esp32
 heltec_wifi_kit_32.build.core=esp32
 heltec_wifi_kit_32.build.variant=heltec_wifi_kit_32
-heltec_wifi_kit_32.build.board=HELTEC_WIFI_KIT_32
+heltec_wifi_kit_32.build.board=heltec_wifi_kit_32
```
Comment (P-R-O-C-H-Y) [inline boards.txt:21231]: The build.board has to be uppercase, please revert this change in all boards.

File: boards.txt [hunk 7]
```diff
@@ -21786,6 +21776,1064 @@ heltec_wifi_lora_32_V3.menu.EraseFlash.all.upload.erase_cmd=-e
...
+heltec_wireless_stick_V3.menu.SLOW_CLK_TPYE.0.build.SLOW_CLK_TPYE=0
+heltec_wireless_stick_V3.menu.SLOW_CLK_TPYE.1=External 32K
+heltec_wireless_stick_V3.menu.SLOW_CLK_TPYE.1.build.SLOW_CLK_TPYE=1
+
+heltec_wireless_stick_V3.build.defines=-D{build.band} -DMCU_ESP32_S3 -DHELTEC_BOARD=31 -DWIRELESS_STICK_V3 -DSLOW_CLK_TPYE={build.SLOW_CLK_TPYE} -DRADIO_CHIP_SX1262  -DLoRaWAN_DEBUG_LEVEL={build.LoRaWanDebugLevel} -DACTIVE_REGION=LORAMAC_{build.band} -DLORAWAN_PREAMBLE_LENGTH={build.LORAWAN_PREAMBLE_LENGTH} -DLORAWAN_DEVEUI_AUTO={build.LORAWAN_DEVEUI_AUTO} -D{build.board}
+
+heltec_wireless_stick_V3.menu.EraseFlash.none=Disabled
+heltec_wireless_stick_V3.menu.EraseFlash.none.upload.erase_cmd=
+heltec_wireless_stick_V3.menu.EraseFlash.all=Enabled
+heltec_wireless_stick_V3.menu.EraseFlash.all.upload.erase_cmd=-e
```
Comment (P-R-O-C-H-Y) [inline boards.txt:21925]: Please add an empty line before board ending (####)

File: boards.txt [hunk 8]
```diff
@@ -21786,6 +21776,1064 @@ heltec_wifi_lora_32_V3.menu.EraseFlash.all.upload.erase_cmd=-e
...
+heltec_wireless_shell_V3.menu.SLOW_CLK_TPYE.0.build.SLOW_CLK_TPYE=0
+heltec_wireless_shell_V3.menu.SLOW_CLK_TPYE.1=External 32K
+heltec_wireless_shell_V3.menu.SLOW_CLK_TPYE.1.build.SLOW_CLK_TPYE=1
+
+heltec_wireless_shell_V3.build.defines=-D{build.band} -DMCU_ESP32_S3 -DHELTEC_BOARD=33 -DWIRELESS_SHELL_V3 -DSLOW_CLK_TPYE={build.SLOW_CLK_TPYE} -DRADIO_CHIP_SX1262  -DLoRaWAN_DEBUG_LEVEL={build.LoRaWanDebugLevel} -DACTIVE_REGION=LORAMAC_{build.band} -DLORAWAN_PREAMBLE_LENGTH={build.LORAWAN_PREAMBLE_LENGTH} -DLORAWAN_DEVEUI_AUTO={build.LORAWAN_DEVEUI_AUTO} -D{build.board}
+
+heltec_wireless_shell_V3.menu.EraseFlash.none=Disabled
+heltec_wireless_shell_V3.menu.EraseFlash.none.upload.erase_cmd=
+heltec_wireless_shell_V3.menu.EraseFlash.all=Enabled
+heltec_wireless_shell_V3.menu.EraseFlash.all.upload.erase_cmd=-e
```
Comment (P-R-O-C-H-Y) [inline boards.txt:22223]: Please add an empty line before board definition ending (####)

File: boards.txt [hunk 9]
```diff
@@ -21786,6 +21776,1064 @@ heltec_wifi_lora_32_V3.menu.EraseFlash.all.upload.erase_cmd=-e
...
+heltec_capsule_sensor_V3.serial.disableDTR=false
+heltec_capsule_sensor_V3.serial.disableRTS=false
+
+heltec_capsule_sensor_V3.build.tarch=xtensa
+heltec_capsule_sensor_V3.build.bootloader_addr=0x0
+heltec_capsule_sensor_V3.build.target=esp32s3
+heltec_capsule_sensor_V3.build.mcu=esp32s3
+heltec_capsule_sensor_V3.build.core=esp32
+heltec_capsule_sensor_V3.build.variant=heltec_capsule_sensor_V3
+heltec_capsule_sensor_V3.build.board=heltec_capsule_sensor_V3
```
Comment (P-R-O-C-H-Y) [inline boards.txt:22253]: build.board needs to be uppercase

File: boards.txt [hunk 10]
```diff
@@ -21786,6 +21776,1064 @@ heltec_wifi_lora_32_V3.menu.EraseFlash.all.upload.erase_cmd=-e
...
+
+heltec_capsule_sensor_V3.build.defines=-D{build.band} -DMCU_ESP32_S3 -DHELTEC_BOARD=50 -DCAPSULE_SENSOR_V3 -DSLOW_CLK_TPYE={build.SLOW_CLK_TPYE} -DRADIO_CHIP_SX1262  -DLoRaWAN_DEBUG_LEVEL={build.LoRaWanDebugLevel} -DACTIVE_REGION=LORAMAC_{build.band} -DLORAWAN_PREAMBLE_LENGTH={build.LORAWAN_PREAMBLE_LENGTH} -DLORAWAN_DEVEUI_AUTO={build.LORAWAN_DEVEUI_AUTO} -D{build.board} -DNLOG_LOCAL_LEVEL={build.NetworkLogLevel} 
+
+heltec_capsule_sensor_V3.menu.EraseFlash.none=Disabled
+heltec_capsule_sensor_V3.menu.EraseFlash.none.upload.erase_cmd=
+heltec_capsule_sensor_V3.menu.EraseFlash.all=Enabled
+heltec_capsule_sensor_V3.menu.EraseFlash.all.upload.erase_cmd=-e
+
+
+
```
Comment (P-R-O-C-H-Y) [inline boards.txt:22385]: Remove 2 unnecessary empty lines.
```suggestion
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@Heltec-Aaron-Lee PTAL on my comments.
Few point to be changed in almost all boards definition ( I did not cover everything in the review):
- all build.board names must be uppercase
- check formatting to make everything clear, remove or add empty lines please.

When fixed, ping me to review again.

**Comment by Heltec-Aaron-Lee** [comment]
@P-R-O-C-H-Y 
Hi, thank you very much for taking the time to review. I have corrected the issues you mentioned, please review again.

File: boards.txt [hunk 11]
```diff
@@ -22020,259 +23054,257 @@ heltec_wireless_stick_lite.menu.LoRaWanDebugLevel.2.build.LoRaWanDebugLevel=2
...
+heltec_ht_de01.menu.einksize.0=eink_150
+heltec_ht_de01.menu.einksize.0.build.einksize=150;
+heltec_ht_de01.menu.einksize.1=eink_154
+heltec_ht_de01.menu.einksize.1.build.einksize=154;
+heltec_ht_de01.menu.einksize.2=eink_213
+heltec_ht_de01.menu.einksize.2.build.einksize=213;
+heltec_ht_de01.menu.einksize.3=eink_290
+heltec_ht_de01.menu.einksize.3.build.einksize=290;
+
+heltec_ht_de01.build.defines=   -DEINK={build.einksize}   -D{build.board}
```
Comment (P-R-O-C-H-Y) [inline boards.txt:23308]: The Heltec E-Ink driver is missing the Eraseflash menu. Please add it.
```suggestion

heltec_ht_de01.menu.EraseFlash.none=Disabled
heltec_ht_de01.menu.EraseFlash.none.upload.erase_cmd=
heltec_ht_de01.menu.EraseFlash.all=Enabled
heltec_ht_de01.menu.EraseFlash.all.upload.erase_cmd=-e
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
Looks good, just one fix needed :)
PTAL @Heltec-Aaron-Lee

**Comment by Heltec-Aaron-Lee** [comment]
I apologize for my carelessness... I have corrected this issue, please review it again
@P-R-O-C-H-Y

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Pull request merged**